hummingbot/connector/exchange/hotbit/hotbit_api_user_stream_data_source.py

```diff
@@ -0,0 +1,99 @@
+import asyncio
+from typing import TYPE_CHECKING, List, Optional
+
+from hummingbot.connector.exchange.hotbit import hotbit_constants as CONSTANTS
+from hummingbot.connector.exchange.hotbit.hotbit_auth import HotbitAuth
+from hummingbot.connector.utils import split_hb_trading_pair
+from hummingbot.core.data_type.user_stream_tracker_data_source import UserStreamTrackerDataSource
+from hummingbot.core.web_assistant.connections.data_types import WSJSONRequest
+from hummingbot.core.web_assistant.web_assistants_factory import WebAssistantsFactory
+from hummingbot.core.web_assistant.ws_assistant import WSAssistant
+from hummingbot.logger import HummingbotLogger
+
+if TYPE_CHECKING:
+    from hummingbot.connector.exchange.hotbit.hotbit_exchange import HotbitExchange
+
+
+class HotbitAPIUserStreamDataSource(UserStreamTrackerDataSource):
+
+    LISTEN_KEY_KEEP_ALIVE_INTERVAL = 1800  # Recommended to Ping/Update listen key to keep connection alive
+    HEARTBEAT_TIME_INTERVAL = 30.0
+
+    _logger: Optional[HummingbotLogger] = None
+
+    def __init__(self,
+                 auth: HotbitAuth,
+                 trading_pairs: List[str],
+                 connector: 'HotbitExchange',
+                 api_factory: WebAssistantsFactory,
+                 domain: str = CONSTANTS.DEFAULT_DOMAIN):
+        super().__init__()
+        self._auth: HotbitAuth = auth
+        self._current_listen_key = None
+        self._domain = domain
+        self._api_factory = api_factory
+        self._trading_pairs = trading_pairs
+        self._connector = connector
+
+        self._listen_key_initialized_event: asyncio.Event = asyncio.Event()
+        self._last_listen_key_ping_ts = 0
+
+    async def _connected_websocket_assistant(self) -> WSAssistant:
+        """
+        Creates an instance of WSAssistant connected to the exchange
+        """
+        ws: WSAssistant = await self._api_factory.get_ws_assistant()
+        await ws.connect(ws_url=CONSTANTS.WSS_URL, ping_timeout=CONSTANTS.WS_HEARTBEAT_TIME_INTERVAL)
+        sign = self._auth.sign(params={})
+        payload = {
+            "method": "server.auth",
+            "params": [self._auth.api_key, sign],
+            "id": 1
+        }
+        auth_request: WSJSONRequest = WSJSONRequest(payload=payload)
+        await ws.send(auth_request)
+        return ws
+
+    async def _subscribe_channels(self, websocket_assistant: WSAssistant):
+        """
+        Subscribes to the trade events and diff orders events through the provided websocket connection.
+
+        Hotbit does not require any channel subscription.
```

Comment:
is this correct? if so, why are you sending the request?

Reply:
This is not correct. I forgot to update it when I duplicated the file from binance.Fix it on the latest patch.